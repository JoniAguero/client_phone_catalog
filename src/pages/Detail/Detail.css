.container-detail-phone {
  margin-top: 25px;
}
.container-detail-phone-loading{
  margin-top: 100px;
}