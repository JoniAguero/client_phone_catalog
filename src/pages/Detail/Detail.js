import React, { useEffect, useState } from "react";
import { useSelector } from "react-redux";
import DetailPhone from "../../components/DetailPhone/DetailPhone";
import Loading from "../../components/Loading/Loading";
import PlaceholderText from "../../components/Loading/PlaceholderText";
import NoDataFound from "../../components/NoDataFound";
import "./Detail.css";

export const Detail = () => {

  const [loading, setLoading] = useState(true)
  const { phones } = useSelector( state => state )
  const phoneSelected = phones.phoneSelected;
  
  useEffect(() => {
    setTimeout(() => {
      setLoading(false)
    }, 1000);
  }, [])

  if ( loading ) {
    return (
      <div className="container-detail-phone-loading">
        <Loading repetitions={1}>
          <PlaceholderText />
        </Loading>
      </div> 
    );
  }

  return (
    <div className="container-detail-phone">
      { 
        !phoneSelected ? <NoDataFound /> : 
        <DetailPhone phone={phoneSelected}/> 
      }

    </div>
  );
}