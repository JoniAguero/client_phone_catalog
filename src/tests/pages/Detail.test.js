import React from 'react'
import { mount } from 'enzyme'
import { Provider } from 'react-redux'

import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import '@testing-library/jest-dom'

import Detail from '../../pages/Detail'

jest.mock('../../redux/actions/phonesActions', () => ({
  getPhones: jest.fn(),
}))

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initialState = {
  phones: {
    list: [],
    phoneSelected: null
  },
  errors: {
    error: {}
  }
}

const store = mockStore(initialState)
store.dispatch = jest.fn()

const wrapper = mount(
  <Provider store={store}>
    <Detail />
  </Provider>
)

describe('Detail Page: ', () => {
  test('should be displayed correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
