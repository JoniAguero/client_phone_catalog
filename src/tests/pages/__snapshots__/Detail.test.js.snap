// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Page:  should be displayed correctly 1`] = `
<div
  className="container-detail-phone-loading"
>
  <div
    className="container-loading"
  >
    <div
      key="0"
    >
      <div
        className="ui fluid placeholder"
      >
        <div
          className="image header"
        >
          <div
            className="line"
          />
          <div
            className="line"
          />
        </div>
        <div
          className="paragraph"
        >
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
          <div
            className="line"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
